auth/realm/base/src/main/java/org/wildfly/security/auth/realm/AggregateSecurityRealm.java

```diff
@@ -88,12 +101,18 @@ public RealmIdentity getRealmIdentity(final Evidence evidence) throws RealmUnava
     public RealmIdentity getRealmIdentity(final Principal principal) throws RealmUnavailableException {
         boolean ok = false;
         final RealmIdentity authenticationIdentity = authenticationRealm.getRealmIdentity(principal);
+
+        Principal authorizationPrincipal = principal;
+        if(principalTransformer != null) {
+            authorizationPrincipal = principalTransformer.apply(authorizationPrincipal);
+            if (authorizationPrincipal == null) return null;
```

Comment:
Need to be a bit careful here since ```getRealmIdentity``` should only return a non-null value. I think we should throw an ```IllegalStateException``` here if we have a ```null``` value at this point.

Reply:
Fixed